src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (target.hidden !== true && target instanceof Herbivore) {
```

Comment:
Checking for 'target.hidden !== true' is redundant since 'true' is the only truthy value 'hidden' can have. You can simplify this check to just 'if (!target.hidden && target instanceof Herbivore)'.